### PR TITLE
Allow expected failures in dependents

### PR DESCRIPTION
**Pull request by forivall**
This change should allow https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61001 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61620 to pass CI

### CODE REVIEW

**Comment by andrewbranch** [comment]
Ah, you managed to catch both me and @sandersn on vacation. I believe @sandersn is back tomorrow-ish and would know more about this than I would. Also cc @jakebailey who’s on DT this week.

**Comment by forivall** [comment]
Awesome, no worries. Yeah, I have no idea if this is the correct solution. I just want to get those lodash changes in, and so looking for whatever way to get it done, and this conceptually might do the job, so why not.

Thanks! I'll ping Jake on the DT repo on those issues.

File: packages/dtslint-runner/src/main.ts [hunk 1]
```diff
@@ -161,16 +161,16 @@ export async function runDTSLint({
...
   if (onlyRunAffectedPackages) {
     return undefined;
   }
 
   return new Set(
     (readFileSync(joinPaths(__dirname, "../expectedFailures.txt"), "utf8") as string)
       .split("\n")
-      .filter(Boolean)
       .map((s) => s.trim())
+      .filter(onlyRunAffectedPackages ? (line) => line && dependents.includes(line) : Boolean)
```
Comment (jakebailey) [inline packages/dtslint-runner/src/main.ts:173]: This function returns early if `onlyRunAffectedPackages` is true, so I'm not sure that this is having the affect you intend in that case.

Comment (jakebailey) [reply packages/dtslint-runner/src/main.ts:173]: Yeah, this PR is effectively a noop (besides running this after `trim`), as `onlyRunAffectedPackages ` is guaranteed to be false by this point, and therefore `Boolean` will be chosen.

File: packages/dtslint-runner/src/main.ts [hunk 2]
```diff
@@ -161,16 +161,16 @@ export async function runDTSLint({
   return allFailures.length;
 }
 
-function getExpectedFailures(onlyRunAffectedPackages: boolean) {
+function getExpectedFailures(onlyRunAffectedPackages: boolean, dependents: readonly string[]) {
   if (onlyRunAffectedPackages) {
     return undefined;
   }
```
Comment (forivall) [inline packages/dtslint-runner/src/main.ts:167]: These lines were supposed to be removed. Idk why I missed that in the commit, but I'll update this.

File: packages/dtslint-runner/src/main.ts (hunk 1 above)
Comment (forivall) [reply packages/dtslint-runner/src/main.ts:173]: Ah, yeah, that early exit conditional was supposed to be removed.

**Review by sandersn — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]
I was going to try and make a test for this, but the runner doesn't have any tests anyway. I'll merge and release it, then; I'm pretty sure this is right.

**Pull request merged**